scheduler/pkg/kafka/gateway/worker.go

```diff
@@ -343,14 +343,19 @@ func (iw *InferWorker) restRequest(ctx context.Context, job *InferWork, maybeCon
 		return iw.produce(ctx, job, iw.topicNamer.GetModelErrorTopic(), b, true, nil)
 	}
 
-	return iw.produce(
+	err = iw.produce(
```

Comment:
Just a note that if this produce fails, it's likely that the produce in the error handling will also fail. But let's keep it like this, because there are edge cases (like if someone deleted the output topic) where it's worth exposing that on the error topic.